flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -193,6 +197,11 @@ public SchedulerBase(
 		this.failoverTopology = executionGraph.getFailoverTopology();
 
 		this.inputsLocationsRetriever = new ExecutionGraphToInputsLocationsRetrieverAdapter(executionGraph);
+
+		// Use the counter from execution graph to avoid modifying execution graph interfaces
+		// Can be a new SimpleCounter created here after the legacy scheduler is removed.
+		this.numberOfRestartsCounter = executionGraph.getNumberOfRestartsCounter();
+		jobManagerJobMetricGroup.meter(NUMBER_OF_RESTARTS, new MeterView(numberOfRestartsCounter));
```

Comment:
Yes the rate is awkward if the event happens in a very low frequency.
I think a counter `numberOfRestarts` is needed to enable users to build alerts in a more flexible way.
And the question is: Whether to introduce a meter `numberOfRestartsPerSecond`?
- Pros: The meter enables users to build alerts for restarts even if their monitoring system does not support variations of values. 
- Cons: The integral of rate value is not accurate so that users cannot use it to build reliable alerts other than ">0". This is limited by the timespan used to sample metrics in Flink, as well as in the external metric collecting system.

Reply:
cc: @stevenzwu

Reply:
I agree that the meter doesn't make sense; the volume of events is simply to low.

You'd want something it to work on hours I suppose, which is possible to implement (it's just a slightly modified MeterView, see below for the differences), but we'd technically violate the meter interface as it explicitly says that rates are per second.

And I am wondering whether it wouldn't be perfectly fine to push this responsibility to the user, given that all time-series databases support queries that can derive the rate in whatever granularity they desire.
That's my overall stance on metrics anyway; expose the basic raw data, and have users build more sophisticated things on top, with the only exception being things we want in the WebUI.



How to implement a HourMeter:
- take the `MeterView`
- pin the values array to 60 values
- don't divide by the timeSpan when calculating the rate
- only act on every 12th update() call

Reply:
Thanks @zentol for the detailed explanation! 

I agree that expose such an low frequency and inaccurate meter is not good. 
But I'm not sure whether it's good to introduce HourMeter since, as you said, it violates the meter interface. And there can also be requirement for MinuteMeter if we take it this way.

Maybe we should revive the [discussion](https://lists.apache.org/thread.html/6ed95eb6a91168dba09901e158bc1b6f4b08f1e176db4641f79de765@%3Cdev.flink.apache.org%3E). The previous conclusion was that a meter should be added. And we need to updates our latest thoughts to see how we can reach a consensus.